test/server.js

```diff
@@ -0,0 +1,250 @@
+const assert = require('assert').strict;
+const { AssertionError } = require('assert');
+const expect = require('expect');
+const app = require('express')();
+const server = require('http').Server(app);
+const io = require('socket.io')(server);
+const toJsonSchema = require('@openapi-contrib/openapi-schema-to-json-schema');
+const Ajv = require('ajv');
+const bodyParser = require('body-parser');
+const { isValidUUIDV4 } = require('is-valid-uuid-v4');
+
+const metricsApi = require('./openapi.json');
+const harFixtures = require('./payloads');
+
+const port = 3000;
+const baseLogUrl = 'https://docs.example.com';
+
+let bufferedSet = 1;
+
+app.use(bodyParser.json({}));
+app.use((req, res, next) => {
+  res.io = io;
+  next();
+});
+
+app.use((req, res, next) => {
+  const auth = req.header('authorization');
+  if (!auth) {
+    return res.status(401).json({
+      message: 'You must pass in an API key',
+    });
+  }
+
+  const userAgent = req.header('user-agent');
+  if (!userAgent) {
+    return res.status(400).json({
+      message: 'No user agent was supplied in a request from the SDK.',
+    });
+  } else if (userAgent !== 'readmeio/4.0.0') {
```

Comment:
Why does it have to be this exact version? Shouldn't we just enforce that it's readme prefixed and semver format?

Reply:
The idea I had here was to enforce that each SDK was sending the right/current version in requests but just checking if it's readme+semver is more sensible so I'll move to that.